packages/gensx-react/src/hooks/use-gensx.ts

```diff
@@ -0,0 +1,498 @@
+   * Can receive either a string message or a parsed JSON object for structured progress events
+   */
+  onProgress?: (message: string | any) => void;
+
+  /**
+   * Callback fired for each output chunk (streaming mode only)
+   */
+  onOutput?: (chunk: string) => void;
+
+  /**
+   * Callback fired when workflow completes
+   */
+  onComplete?: (output: TOutput) => void;
+
+  /**
+   * Callback fired on error
+   */
+  onError?: (error: string) => void;
+
+  /**
+   * Callback fired for any event
+   */
+  onEvent?: (event: TEvent) => void;
+}
+
+// Union type for workflow control events (excludes progress and output)
+export type GenSXWorkflowEvent =
+  | GenSXStartEvent
+  | GenSXComponentStartEvent
+  | GenSXComponentEndEvent
+  | GenSXEndEvent
+  | GenSXErrorEvent;
+
+export interface UseGenSXResult<
+  TInputs = any,
+  TOutput = any,
+  TEvent extends GenSXProgressEvent = GenSXProgressEvent,
+> {
+  /** Whether the workflow is currently running */
+  isLoading: boolean;
+
+  /** Whether the workflow is streaming */
+  isStreaming: boolean;
+
+  /** Any error that occurred */
+  error: string | null;
+
+  /** The final output (collection mode) */
+  output: TOutput | null;
+
+  /** All events received */
+  events: GenSXEvent[];
+
+  /** All events received */
+  workflowEvents: GenSXEvent[];
+
+  /** Progress events */
+  progressEvents: TEvent[];
+
+  /** Output events */
+  outputEvents: GenSXOutputEvent[];
+
+  /** Run workflow in collection mode (returns final output) */
+  run: (inputs: TInputs) => Promise<TOutput | null>;
+
+  /** Run workflow in streaming mode */
+  stream: (inputs: TInputs) => Promise<void>;
+
+  /** Stop the current workflow */
+  stop: () => void;
+
+  /** Clear all state */
+  clear: () => void;
+}
+
+/**
+ * Hook for interacting with GenSX workflows via your API endpoint
+ *
+ * Matches the GenSX Client interface for easy passthrough implementations
+ *
+ * @example
+ * ```tsx
+ * // Configure defaults
+ * const gensx = useGenSX({
+ *   endpoint: '/api/gensx',
+ *   defaultConfig: {
+ *     org: 'my-org',
+ *     project: 'my-project',
+ *     environment: 'production'
+ *   },
+ *   onOutput: (chunk) => console.log(chunk),
+ *   onComplete: (output) => console.log('Done:', output)
+ * });
+ *
+ * // Collection mode - wait for final output
+ * const result = await gensx.run('ChatWorkflow', {
+ *   inputs: { userMessage: 'Hello' }
+ * });
+ *
+ * // Streaming mode - get chunks as they arrive
+ * await gensx.stream('ChatWorkflow', {
+ *   inputs: { userMessage: 'Tell me a story' }
+ * });
+ *
+ * // Override defaults for specific call
+ * await gensx.run('ChatWorkflow', {
+ *   org: 'different-org',
+ *   project: 'different-project',
+ *   inputs: { userMessage: 'Hello from different org' }
+ * });
+ * ```
+ */
+export function useWorkflow<
+  TInputs = any,
+  TOutput = any,
+  // This should actually be BaseProgressEvent
+  TEvent extends GenSXProgressEvent = GenSXProgressEvent,
```

Comment:
What do we need this generic for? If it isn't essential, might be best to remove.